pkg/client/results/junit.go

```diff
@@ -18,25 +18,104 @@ package results
+	// JUnitStdoutKey is the key in the Items.Details map for the system-out output.
+	JUnitStdoutKey = "system-out"
 
-	// JunitFailureKey is the key in the Items.Details map for the failure output.
-	JunitFailureKey = "failure"
+	// JUnitFailureKey is the key in the Items.Details map for the failure output.
+	JUnitFailureKey = "failure"
 )
 
-// Filter keeps only the tests that match the predicate function.
-func Filter(predicate func(testCase reporters.JUnitTestCase) bool, testSuite reporters.JUnitTestSuite) []reporters.JUnitTestCase {
-	out := make([]reporters.JUnitTestCase, 0)
+// junitResult is a wrapper around the suite[s] which enable results to
+// be either a single suite or a collection of suites. For instance,
+// e2e tests (which use the onsi/ginkgo reporter) report a single, top-level
+// testsuite whereas other tools report a top-level testsuites object
+// which may have 1+ testsuite children. Only one of the fields should be
+// set, not both.
+type junitResult struct {
+	suites JUnitTestSuites
+}
+
+// UnmarshalXML will unmarshal the document into either the 'Suites'
+// field of the JUnitResult. If a single testsuite is found (instead of a testsuites object)
+// it is unmarshalled and then added into the set of JUnitResult.Suites.Suites
+func (j *junitResult) UnmarshalXML(d *xml.Decoder, start xml.StartElement) error {
+	var returnErr error
+	if start.Name.Local == "testsuites" {
+		returnErr = d.DecodeElement(&j.suites, &start)
+	} else {
+		var s JUnitTestSuite
+		returnErr = d.DecodeElement(&s, &start)
+		j.suites = JUnitTestSuites{Suites: []JUnitTestSuite{s}}
+	}
+	for i := range j.suites.Suites {
+		if j.suites.Suites[i].Name == "" {
+			j.suites.Suites[i].Name = fmt.Sprintf("testsuite-%03d", i+1)
+		}
+	}
+	return returnErr
+}
+
+// JUnitTestSuites is a collection of JUnit test suites.
+type JUnitTestSuites struct {
+	XMLName xml.Name         `xml:"testsuites"`
+	Suites  []JUnitTestSuite `xml:"testsuite"`
+}
+
+// JUnitTestSuite is a single JUnit test suite which may contain many
+// testcases.
+type JUnitTestSuite struct {
+	XMLName    xml.Name        `xml:"testsuite"`
+	Tests      int             `xml:"tests,attr"`
+	Failures   int             `xml:"failures,attr"`
+	Time       float64         `xml:"time,attr"`
+	Name       string          `xml:"name,attr"`
+	Properties []JUnitProperty `xml:"properties>property,omitempty"`
+	TestCases  []JUnitTestCase `xml:"testcase"`
+}
+
+// JUnitTestCase is a single test case with its result.
+type JUnitTestCase struct {
+	XMLName     xml.Name             `xml:"testcase"`
+	Classname   string               `xml:"classname,attr"`
+	Name        string               `xml:"name,attr"`
+	Time        string               `xml:"time,attr"`
+	SkipMessage *JUnitSkipMessage    `xml:"skipped,omitempty"`
+	Failure     *JUnitFailureMessage `xml:"failure,omitempty"`
+	SystemOut   string               `xml:"system-out,omitempty"`
+	SystemErr   string               `xml:"system-err,omitempty"`
```

Comment:
New field; not really used right now but could be. Didnt add this into the metadata but could.

Reply:
Update: Went ahead and added the processing to support this now too. Just seems too intuitive a field to not include